pageserver/src/remote_storage/storage_sync.rs

```diff
@@ -852,18 +864,28 @@ async fn update_local_metadata(
 
     if local_lsn < Some(remote_lsn) {
         info!("Updating local timeline metadata from remote timeline: local disk_consistent_lsn={local_lsn:?}, remote disk_consistent_lsn={remote_lsn}");
-
-        let remote_metadata_bytes = remote_metadata
-            .to_bytes()
-            .context("Failed to serialize remote metadata to bytes")?;
-        fs::write(&local_metadata_path, &remote_metadata_bytes)
-            .await
-            .with_context(|| {
-                format!(
-                    "Failed to write remote metadata bytes locally to path '{}'",
-                    local_metadata_path.display()
-                )
-            })?;
+        // clone because spawn_blocking requires static lifetime
+        let cloned_metadata = remote_metadata.to_owned();
+        let ZTenantTimelineId {
+            tenant_id,
+            timeline_id,
+        } = sync_id;
+        tokio::task::spawn_blocking(move || {
```

Comment:
@SomeoneToIgnore what do you think about moving metadata update to layered repo's apply_sync_status_update?

Reply:
It started to take longer than I expected, I'll do that separately